src/Algebra/Graph/Acyclic/AdjacencyMap.hs

```diff
@@ -0,0 +1,112 @@
+module Algebra.Graph.Acyclic.AdjacencyMap where
+
+import qualified Algebra.Graph.AdjacencyMap as AM
+import qualified Algebra.Graph.AdjacencyMap.Algorithm as AMA
+import qualified Algebra.Graph.AdjacencyMap.Internal as AMI
+import qualified Algebra.Graph.NonEmpty.AdjacencyMap as NonEmpty
+import Control.Monad.Trans.Class (lift)
+import Control.Monad.Trans.State.Strict
+import Data.Map (Map, (!))
+import qualified Data.Map.Strict as Map
+import qualified Data.Set as Set
+
+newtype AdjacencyMap a = AAM
+  { aam :: AM.AdjacencyMap a
+  } deriving (Show)
+
+consistent :: Ord a => AdjacencyMap a -> Bool
+consistent (AAM m) = AMI.consistent m && AMA.isAcyclic m
+
+type AdjacencyMapP a = AdjacencyMap (Int, a)
+
+empty :: AdjacencyMap a
+empty = AAM AM.empty
+
+vertex :: a -> AdjacencyMap a
+vertex x = AAM $ AM.vertex x
+
+vertices :: (Ord a) => [a] -> AdjacencyMap a
+vertices xs = AAM $ AM.vertices xs
+
+overlayD ::
+     (Ord a, Ord b)
+  => AdjacencyMap a
+  -> AdjacencyMap b
+  -> AdjacencyMap (Either a b)
+overlayD (AAM a) (AAM b) = AAM (AM.overlay (AM.gmap Left a) (AM.gmap Right b))
+
+connectD ::
+     (Ord a, Ord b)
+  => AdjacencyMap a
+  -> AdjacencyMap b
+  -> AdjacencyMap (Either a b)
+connectD (AAM a) (AAM b) = AAM (AM.connect (AM.gmap Left a) (AM.gmap Right b))
+
+transitiveClosure :: Ord a => AdjacencyMap a -> AdjacencyMap a
+transitiveClosure (AAM x) = AAM (AM.transitiveClosure x)
+
+overlay :: (Ord a) => AdjacencyMap a -> AdjacencyMap a -> AdjacencyMap a
+overlay (AAM x) (AAM y) = AAM (AM.overlay x y)
+
+connect :: (Ord a) => AdjacencyMap a -> AdjacencyMap a -> AdjacencyMap a
+connect (AAM x) (AAM y) =
+  AAM . AMI.AM $
+  Map.unionsWith Set.union $
+  AM.adjacencyMap x :
+  AM.adjacencyMap y :
+  [ Map.fromSet
+      (\k -> Set.filter (< k) . Map.keysSet $ AM.adjacencyMap y)
+      (Map.keysSet $ AM.adjacencyMap x)
+  ]
+
+-- reverse and Map.keys results in an ascending list
+topSort :: (Ord a) => AdjacencyMap a -> [a]
+topSort (AAM (AMI.AM m)) = reverse $ Map.keys m
+
+-- scc
+scc :: (Ord a) => AM.AdjacencyMap a -> AdjacencyMapP (NonEmpty.AdjacencyMap a)
+scc am = unsafeToAcyclic $ AMA.scc am
+
+unsafeToAcyclic :: (Ord a) => AM.AdjacencyMap a -> AdjacencyMapP a
```

Comment:
Is this function supposed to rely on partial order? It doesn't match the required API.

As for an unsafe way to create an acyclic graph from a cyclic graph, we can just use `AAM`, i.e. the constructor itself.

Reply:
This function is not for the partial order API. That is yet to be created.

All the vertices in `AM.AdjacencyMap` need not follow the proper ordering as required by the Invariant.
This function is used to induce a new ordering to the vertices so that the invariant is not broken.

Reply:
I'm afraid I'm lost. I don't understand why any extra invariants are needed on the ordering of vertices.

Reply:
I apologize, my bad! Maybe I was thinking far too ahead making things complicated.